admin-dev/themes/new-theme/js/pages/product/components/combination-modal/CombinationModal.vue

```diff
@@ -182,27 +268,92 @@
 
         // This closes the modal which is conditioned to the presence of this value
         this.selectedCombinationId = null;
+
+        // Reset history on close
+        this.combinationsHistory = [];
       },
       showPrevious() {
         if (this.previousCombinationId !== null) {
-          this.selectedCombinationId = this.previousCombinationId;
+          if (this.isFormUpdated) {
```

Comment:
This piece of code is used three times, it could have been turned into a `selectCombination` method